### PR TITLE
[attribute-behavior] Snapshot test

### PR DESCRIPTION
**Pull request by acdlite**
Same data as the attribute table, but in the form of a snapshot. Uses puppeteer to run headless Chromium.

### CODE REVIEW

**Comment by acdlite** [comment]
If we're not ready to check in the actual snapshot file yet, because we aren't happy with the output, I would prefer to check in the rest of the changes in this PR (like extracting the config into a separate module) so I don't have to keep rebasing whenever we change the table.

**Comment by nhunzaker** [comment]
Ah, this is super cool. I was wondering if we'd ever do headless testing of this stuff.

**Comment by gaearon** [comment]
@nhunzaker Do you want to help out reviewing this PR?

**Comment by acdlite** [comment]
Rebased

File: src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      let didStartTest = false;
+      child.stdout.on('data', data => {
+        const str = data.toString('utf8');
+        if (str.includes('Failed')) {
+          console.error('Failure on CRA startup');
+          done();
+          return;
+        }
+
+        if (!didStartTest && str.includes('http://localhost:9292/')) {
```
Comment (acdlite) [inline src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js]: There's probably a better way to do this but idk what it is :D

Comment (acdlite) [reply src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js]: What I really want is a callback that says "Hey CRA is serving your app now, go ahead and run your tests."

**Comment by gaearon** [comment]
How does this work? Does it build the bundles first?

**Comment by acdlite** [comment]
Nah, just runs CRA dev server. But you're right, would probably be easier to just build the bundle and then serve a static file. I think maybe puppeteer lets you give it a file.

**Comment by gaearon** [comment]
The thing that bothers me is that this doesn't solve the warning deduplication problem (afaik). I always edit the bundle to remove `warnedProperties[` and `warnedProperties$1[` assignments because they make the warning reporting look wrong. But it seems like snapshot test would fall to this problem.

**Comment by acdlite** [comment]
I see. Could we add a secret feature flag?

**Comment by gaearon** [comment]
¯\_(ツ)_/¯

File: src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js (hunk 1 above)
Comment (nhunzaker) [reply src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js]: I'd vote do a production build and run at as a static for serve, if possible.

Comment (nhunzaker) [reply src/renderers/dom/shared/__tests__/DOMAttributeBehavior-test.js]: * that way you have direct control of setup and teardown without any ipc

**Comment by gaearon** [comment]
Superseded by https://github.com/facebook/react/pull/10595.
Do you think we should still make a script that presses "Save" via puppeteer?

**Pull request closed without merging**